Add link to Spezi in README

vishnuravi opened the pull request:
<!--

This source file is part of the CardinalKit open-source project

SPDX-FileCopyrightText: 2022 CardinalKit and the project authors (see CONTRIBUTORS.md)

SPDX-License-Identifier: MIT

-->

# Add link to Spezi in README

## :recycle: Current situation & Problem
CardinalKit iOS is now in maintenance mode and new projects should try [Stanford Spezi](https://github.com/StanfordSpezi).

### Code of Conduct & Contributing Guidelines 

By submitting creating this pull request, you agree to follow our [Code of Conduct](https://github.com/CardinalKit/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/CardinalKit/.github/blob/main/CONTRIBUTING.md):
- [X] I agree to follow the [Code of Conduct](https://github.com/CardinalKit/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/CardinalKit/.github/blob/main/CONTRIBUTING.md).

PSchmiedmayer commented:
@vishnuravi Thank you for the improvements! I moved the note up to the top of the README, made it more clearer, and also added a link to the Template app. What do you think?

vishnuravi commented:
> @vishnuravi Thank you for the improvements! I moved the note up to the top of the README, made it more clearer, and also added a link to the Template app. What do you think?

This is great, thank you @PSchmiedmayer! Will merge.
